jupyterhub/apihandlers/users.py

```diff
@@ -185,80 +185,77 @@ class UserServerAPIHandler(APIHandler):
 
         options = self.get_json_body()
         yield self.spawn_single_user(user, options=options)
-        status = 202 if user.spawn_pending else 201
+        status = 202 if user.spawner._spawn_pending else 201
         self.set_status(status)
 
     @gen.coroutine
     @admin_or_self
     def delete(self, name):
         user = self.find_user(name)
-        if user.stop_pending:
+        if user.spawner._stop_pending:
             self.set_status(202)
             return
-        if not user.running:
+        if not user.running(''):
             raise web.HTTPError(400, "%s's server is not running" % name)
         # include notify, so that a server that died is noticed immediately
         status = yield user.spawner.poll_and_notify()
         if status is not None:
             raise web.HTTPError(400, "%s's server is not running" % name)
         yield self.stop_single_user(user)
-        status = 202 if user.stop_pending else 204
+        status = 202 if user.spawner._stop_pending else 204
         self.set_status(status)
 
 
-class UserCreateNamedServerAPIHandler(APIHandler):
-    """Create a named single-user server
+class UserNamedServerAPIHandler(APIHandler):
+    """Manage named single-user servers
     
     This handler should be used when c.JupyterHub.allow_named_servers = True
     """
     @gen.coroutine
     @admin_or_self
-    def post(self, name):
+    def post(self, name, server_name):
+        if not self.allow_named_servers:
+            raise web.HTTPError(400, "Named servers are not enabled.")
         user = self.find_user(name)
         if user is None:
             raise web.HTTPError(404, "No such user %r" % name)
-        if user.running:
-            # include notify, so that a server that died is noticed immediately
-            state = yield user.spawner.poll_and_notify()
-            if state is None:
-                raise web.HTTPError(400, "%s's server is already running" % name)
-
+        
         options = self.get_json_body()
-        yield self.spawn_single_user(user, options=options)
-        status = 202 if user.spawn_pending else 201
+        yield self.spawn_single_user(user, server_name, options=options)
+        spawner = user.spawners[server_name]
+        status = 202 if spawner._spawn_pending else 201
         self.set_status(status)
 
-
-class UserDeleteNamedServerAPIHandler(APIHandler):
-    """Delete a named single-user server
-    
-    Expect a server_name inside the url /user/:user/servers/:server_name
-    
-    This handler should be used when c.JupyterHub.allow_named_servers = True
-    """
     @gen.coroutine
     @admin_or_self
     def delete(self, name, server_name):
+        if not self.allow_named_servers:
+            raise web.HTTPError(400, "Named servers are not enabled.")
         user = self.find_user(name)
-        if user.stop_pending:
+        if user is None:
+            raise web.HTTPError(404, "No such user %r" % name)
```

Comment:
Good catch. I know I'm not consistent. I think it's whether I feel like `'username'` quotes seem right at the time. If I have a rule, it would be when it's a referring to the user by name, no quotes, but referring to an unattached username, quotes. I could also use `"%s"`, which would probably be clearer. For example:

> willingc's server has started

and

> No user named "willingc"

I see that the two examples you highlighted here don't even follow that rule, so I'll fix them.